packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -20,7 +20,17 @@ import { AssetManifestBuilder } from '../util/asset-manifest-builder';
 import { determineAllowCrossAccountAssetPublishing } from './util/checks';
 import { publishAssets } from '../util/asset-publishing';
 
-export interface DeployStackResult {
+export type DeployStackResult =
+  // Regular result
+  | RegularDeployStackResult
+  // The stack is currently in a failpaused state, and needs to be rolled back before the deployment
+  | { type: 'failpaused-need-rollback-first'; reason: 'not-norollback' | 'replacement' }
+  // The upcoming change has a replacement, which requires deploying without --no-rollback.
+  | { type: 'replacement-requires-norollback' }
+  ;
```

Comment:
Not super important, but this feels to be less that we have a "regular" deploy stack result and other "irregular" results, but we have a Success and than 2 different kinds of Failures. 

Either way, let's model the 2 new types as interfaces instead of inline.